### DISCUSSION TITLE
Gas Update Times

### DISCUSSION THREAD

**Discussion by Dynamic5912** [BottlecapDave/HomeAssistant-OctopusEnergy#324] [General]
What times are the previous days usage etc. for Gas supposed to update?

Mine seems to update at around 17:30 on a daily basis, meaning my previous days consumptions and costs are out by nearly 18 hours.

Electricity seems to update at a more convenient time.

**Comment by BottlecapDave** [comment]
Hello.

The time at which gas times update varies from user to user (and sometimes day to day) and is at the mercy of Octopus Energy, which I believe is at the mercy of when the smart meter data is available to them. As you've already discovered, this seems to be later than electricity. The only thing that I might be in control of is if the data is available within the dashboard/app and it doesn't appear in Home assistant within the hour. I've updated the documentation to clarify this.

> meaning my previous days consumptions and costs are out by nearly 18 hours

I just want to clarify that by out you mean unavailable rather than the data is incorrect?

**Reply by Dynamic5912** [reply]
Hi Dave,

OK - understood.

By data being "out" - I mean, for example, the Previous Accumulative Consumption (for both kWh and M3) are both showing Saturdays usage, but the Previous Accumulative Cost is showing yesterdays total cost, not Saturday like the usage ones quoted above. It all seems very out of whack.

I also have the Energy Dashboard setup as well, and this is giving different readings altogether for both Gas and Electric (compared to my Octopus Account AND your add-in for those days).

**Reply by BottlecapDave** [reply]
hmm that is very odd, as both sensors rely on the same data. Depending on how your meter natively sends data, kwh/m3 information will be estimated which can be out when compared to OE if the default calorific value is dffferent from the value on your bill. Unfortunately, I have yet to find this information supplied in their API.

Am I correct in assuming you're on the latest version of the plugin (v7.5.2)?

If you are, then can you post the following information targeting the "same" previous day and with any personal identifiable information removed

1) Your OE app/dashboard (with total consumption clearly visible)
2) Your HA energy dashboard, with the total consumption clearly visisble) 
3) The attributes of your previous consumption sensor
4) The attributes of your previous consumption cost sensor

Also any errors that are related to the integration in your HA instance.

**Reply by Dynamic5912** [reply]
Hi Dave,

Apologies for the delayed reply.

So the issue is still occuring - for example for readings showing in the HA Dashbaord today:

**Previous Accumulative Consumption for Gas** was 4.344 kWh
**Previous Accumulative Cost for Gas** was £0.44

If I look at yesterdays ACTUAL readings in the Octopus app:

**Consumption** was 8.18 kWh
**Cost** was £0.30 (on par with what HA is showing)

If I look at the day before (i.e. 2 days ago from today):

**Consumption** was 4.38 kWh (on par with what HA is showing)
**Cost** was £0.11

So it looks as if the previous consumption _cost_ is from the **previous day**, but the previous consumption _usage_ was from **2 days ago** still.
